examples/raytracing/example_raytracing.cpp

```diff
@@ -0,0 +1,224 @@
+/****************************************************************************
+ * Copyright (c) 2012-2021 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+#include <random>
+#include <vector>
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+namespace ArborX
+{
+using Primitives = Kokkos::View<Box *, MemorySpace>;
+using Predicates = Kokkos::View<Ray *, MemorySpace>;
+
+template <>
+struct AccessTraits<Predicates, PredicatesTag>
+{
+  KOKKOS_FUNCTION static std::size_t size(const Predicates &predicates)
+  {
+    return predicates.extent(0);
+  }
+  KOKKOS_FUNCTION static auto get(Predicates const &predicates, std::size_t i)
+  {
+    return attach(intersects(predicates(i)), (int)i);
+  }
+  using memory_space = MemorySpace;
+};
+} // namespace ArborX
+
+struct Callback
+{
+  ArborX::Primitives boxes;
+
+  Kokkos::View<float *, MemorySpace, Kokkos::MemoryTraits<Kokkos::Atomic>>
+      accumulator;
+
+  KOKKOS_FUNCTION float dotproduct(ArborX::Point const &v1,
+                                   ArborX::Point const &v2) const
+  {
+    return v1[0] * v2[0] + v1[1] * v2[1] + v1[2] * v2[2];
+  }
+
+  KOKKOS_FUNCTION float solveQuadratic(const float &b, const float &c) const
```

Comment:
The function does not do what its name suggests. It tries to mix solving quadratic with some kind of intersection related things, making it hard to comprehend.